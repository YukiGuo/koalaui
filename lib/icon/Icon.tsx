import React, { SVGAttributes } from 'react';
import mergeClassName from '../tools/classes';
import './icon.scss';
import './importIcons.js';

interface IconProps extends SVGAttributes<SVGElement> {
  name: string;
}
const Icon: React.FunctionComponent<IconProps> = ({
  className,
  name,
  ...restProps
}) => {
  return (
    <svg
      className={mergeClassName('koala-icon', className)}
      aria-hidden="true"
      {...restProps}
    >
      <use xlinkHref={`#${name}`} />
    </svg>
  );
};
export default Icon;
