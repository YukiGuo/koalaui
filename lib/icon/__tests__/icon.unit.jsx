import React from 'react';
import * as renderer from 'react-test-renderer';
import Icon from '../Icon';

describe('icon', () => {
  it('是个 svg', () => {
    console.log('hhhsvg1111');
    const json = renderer.create(<Icon name="add" />).toJSON();
    console.log('hhhsvg');
    expect(json).toMatchSnapshot();
    console.log('hhhsvg2');
  });
});
