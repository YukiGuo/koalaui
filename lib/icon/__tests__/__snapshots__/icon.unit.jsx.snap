// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon 是个 svg 1`] = `
<svg
  aria-hidden="true"
  className="koala-icon"
>
  <use
    xlinkHref="#add"
  />
</svg>
`;
